mcda/electre/class_assignment.py

```diff
@@ -0,0 +1,203 @@
+from typing import Any, Dict, Union
+
+from mcda.core.aliases import NumericValue
+from mcda.electre.outranking import OutrankingRelation, outranking_relation_marginal
+
+import pandas as pd
+
+
+def assign_tri_rc_class(
+    alternatives: Union[Dict[Any, NumericValue], pd.Series],
+    categories_rank: pd.Series,
+    categories_profiles: pd.Series,
+    outranking_ap: pd.DataFrame,
+    outranking_pa: pd.DataFrame,
+    credibility_ap: pd.DataFrame,
+    credibility_pa: pd.DataFrame,
+) -> pd.Series:
+    """
+    :param alternatives:
+    :param categories_rank:
+    :param categories_profiles:
+    :param outranking_ap:
+    :param outranking_pa:
+    :param credibility_ap:
+    :param credibility_pa:
+    :return:
+    """
+    # sorted categories by ranks - ascending (worst to best)
+    categories = [i for i in categories_rank.sort_values(ascending=False)]
+
+    # list of profiles according to categories
+    profiles = list(sorted(categories_profiles))
+
+    assignments_descending = []
+    assignments_ascending = []
+    for a in alternatives:
+        found_descending = False
+        for p in profiles[len(profiles) - 2:: -1]:
+            p_next = profiles[profiles.index(p) + 1]
+            relation = outranking_relation_marginal(
+                outranking_ap.loc[a][p], outranking_pa.loc[p][a]
+            )
+            if (
+                relation == OutrankingRelation.PQ
+                and credibility_ap.loc[a][p_next] > credibility_pa.loc[p][a]
+            ):
+                p_next = categories_profiles[categories_profiles == p_next].index[0]
+                category = categories_profiles[p_next]
+                assignments_descending.append((a, category))
+                found_descending = True
+                break
+
+        if not found_descending:
+            assignments_descending.append((a, categories[0]))
+
+        found_ascending = False
+        for p in profiles[1:]:
+            p_prev = profiles[profiles.index(p) - 1]
+            relation = outranking_relation_marginal(
+                outranking_pa.loc[p][a], outranking_ap.loc[a][p]
+            )
+            if (
+                relation == OutrankingRelation.PQ
+                and credibility_pa.loc[p_prev][a] > credibility_ap.loc[a][p]
+            ):
+                p_prev = categories_profiles[categories_profiles == p_prev].index[0]
+                category = categories_profiles[p_prev]
+                assignments_ascending.append((a, category))
+                found_ascending = True
+                break
+        if not found_ascending:
+            assignments_ascending.append((a, categories[0]))
+    assignments = pd.Series()
```

Comment:
To get rid of that unnecessary mypy warning:
```suggestion
    assignments = pd.Series([], dtype="float64")
```